vpn_slice/mac.py

```diff
@@ -112,3 +112,23 @@ def add_address(self, device, address):
             # with BSD ifconfig. See example in default vpnc-script:
             #   https://gitlab.com/openconnect/vpnc-scripts/blob/https://gitlab.com/openconnect/vpnc-scripts/blob/921e8760/vpnc-script#L193
             self._ifconfig(device, 'inet', address.ip, address.ip, 'netmask', '255.255.255.255')
+
+
+class MacDomainDNSProvider(DomainDNSProvider):
+    def configure_domain_vpn_dns(self, domains, nameservers):
+        if not os.path.exists('/etc/resolver'):
+            os.makedirs('/etc/resolver')
+        for domain in domains:
+            resolver_file_name = "/etc/resolver/{0}".format(domain)
+            resolver_file = open(resolver_file_name, "a")
```

Comment:
I'm curious why you're opening this file for appending (instead of just for writing, mode `"w"`). If you're expecting to preserve any existing content, something special will need to be done during cleanup to undo just the append. If, on the other hand, you consider it to be an error if the file exists, you probably want to open the file for exclusive writing instead (mode `"x"`). That will raise a `FileExistsError` if the file already exists.

Also, please use the context manager form

```python
with open(resolver_file_name, "w") as resolver_file:
    for nameserver in nameservers:
        ...
```
to ensure proper cleanup even in the face of errors.

Reply:
honestly just opened for appending because it worked. maybe in a future PR I'll see about having check for existing entries but I'm not sure on all the possibilities so I think write mode is ok for now.

Thanks for the context manager, didn't know that.